trace/ptrace/ptrace.go

```diff
@@ -438,6 +438,45 @@ func rangeActualScope(r exptrace.Range) rangeScope {
 	}
 }
 
+// runningGauge tracks maximum value of a gauge and creates a time series that can be rendered.
+type runningGauge struct {
+	// current is the current value of the gauge.
+	current int64
+	// currentTime is the time of the last update.
+	currentTime exptrace.Time
+	// series is a list of points that capture the current value.
+	series []Point
+}
+
+// add adds a value to the gauge at time t.
+// It assumes that t will not decrease with subsequent calls.
+func (rg *runningGauge) add(t exptrace.Time, v int64) {
+	if rg.currentTime == 0 {
+		rg.currentTime = t
+	}
+	switch {
+	case rg.currentTime == 0:
+		rg.currentTime = t
+	case t > rg.currentTime:
+		rg.series = append(rg.series, Point{rg.currentTime, uint64(rg.current)})
+		rg.currentTime = t
+	}
+	rg.current += v
+}
+
+func (rg *runningGauge) finalize() []Point {
```

Comment:
Do we need this? We should be able to just add the point when it occurs, instead of lagging one behind.

I'd expect `add` to be akin to this:

```
func (rg *runningGauge) add(t exptrace.Time, v int64) {
  rg.current += v
  if idx := len(rg.series) - 1; idx >= 0 && rg.series[idx].Time == t {
    rg.series[idx].Value = rg.current
  } else {
    rg.series = append(rg.series, Point{t, rg.current})
  }
}
```

Am I missing something?

Reply:
Indeed, we can update the last point directly. I've updated the code. Thank you for the suggestion!